src/cr/cube/matrix/measure.py

```diff
@@ -2191,52 +2418,74 @@ def _stddev_func(self):
-        return self._counts_are_defined
+        """Whether scalar is defined. Defaults to True, but designed to be overridden"""
+        return True
+
+    @lazyproperty
+    def value(self):
+        raise NotImplementedError(  # pragma: no cover
+            "%s must implement `.value`" % type(self).__name__
+        )
 
 
-class _UnweightedBaseMargin(_BaseMarginal):
-    """The unweighted bases margin (sum of unweighted counts)"""
+class _TableWeightedBase(_BaseTableValue):
+    """The 'table-weighted-base', a scalar base for the whole table
+
+    The table-weighted-base is only defined when both rows and columns dimensions
+    are non-array and is equal to the sum of all counts. It was formerly known as the
+    'table-margin' when it was a scalar (which was also only when neither dimension is
+    an arary).
+    """
 
     @lazyproperty
-    def blocks(self):
-        """List of the 2 1D ndarray "blocks" of the summed count margin.
+    def is_defined(self):
+        """Bool indicating whether the table-margin is defined
 
-        These are the base-values and the subtotals.
+        It is defined (as a scalar) only when available on the weighted_cube_counts
+        (eg neither dimension is an array)
         """
+        return self._cube_measures.weighted_cube_counts.table_base is not None
+
+    @lazyproperty
+    def value(self):
+        """float of the table-margin"""
         if not self.is_defined:
-            raise ValueError("Cannot calculate base across subvariables dimension.")
+            raise ValueError(
+                "Cannot sum across subvariables dimension for table weighted base scalar"
+            )
+        return self._cube_measures.weighted_cube_counts.table_base
 
-        # --- Since we know we're not an array variable, we can trust that
-        # --- the 2D array for the unweighted bases measure is the same in
-        # --- the direction we need. Therefore, we can just grab the first
-        # --- row/column of that.
-        if self.orientation == MO.ROWS:
-            bases = self._second_order_measures.row_unweighted_bases.blocks
-            return [bases[0][0][:, 0], bases[1][0][:, 0]]
-        else:
-            bases = self._second_order_measures.column_unweighted_bases.blocks
-            return [bases[0][0][0, :], bases[0][1][0, :]]
+
+class _TableWeightedBasesRange(_BaseTableValue):
+    """The (unpruned) 'table-weighted-bases-range', a length 2 np.array of min and max.
+
+    The range of the table-weighted-bases *before* pruning, eg removing rows and columns
+    that are hidden or empty.
+    """
 
     @lazyproperty
-    def is_defined(self):
-        """True if counts are defined."""
-        return self._counts_are_defined
+    def value(self):
+        """[min, max] np.float64 ndarray range of the table-weighted-base"""
+        bases = self._cube_measures.weighted_cube_counts.table_bases
+        return np.array([np.min(bases), np.max(bases)])
```

Comment:
Yeah, I've thought about that. Because of what I've worked on, the division of `measures`, `margins` and `table_values` makes sense to me, and I also think it helps with how similar some of the names are (`rows_unweighted_base` and `row_unweighted_bases`). However, it doesn't help that much (a good 75% would still be in `measures`, and `table_values` would be pretty empty). 

Maybe splitting off the significance tests makes sense?